File: AndroidTai/app/src/main/java/com/tai/androidtai/dagger/modules/ApplicationModule.java
package com.tai.androidtai.dagger.modules;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;

@Module
public class ApplicationModule {

    public ApplicationModule() {
    }

    @Provides
    @Singleton
    Scheduler provideScheduler() {
        return AndroidSchedulers.mainThread();
    }

}

File: AndroidTai/app/src/main/java/com/tai/androidtai/dagger/modules/ApplicationModule.kt
package com.tai.androidtai.dagger.modules

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers

@Module
class ApplicationModule {

    @Provides
    @Singleton
    internal fun provideScheduler(): Scheduler {
        return AndroidSchedulers.mainThread()
    }

}
